phone: swipe added, merge latest changes from durka:master

habeIchVergessen opened the pull request:
i guess we need something like an agenda. because i have disabled all your phone stuff.

![swipe_phone](https://f.cloud.github.com/assets/3988506/1790575/c68e588a-6979-11e3-9a98-627814e4fa33.png)

Wallace0 committed:
Fixed issue with findViewByID NPEs and issue with alarm screen still
visible if alarm killed elsewhere

Commit:
Merge remote-tracking branch 'origin/call-answer'

Conflicts:
	src/org/durka/hallmonitor/Functions.java

Commit:
Merge remote-tracking branch 'origin/tidy-up'

Conflicts:
	src/org/durka/hallmonitor/DefaultActivity.java

Commit:
code:
- separate functionality (info, phone) in layouts and toggle visibility
- added swipe to phone handler
- minor bug fixes (screen off after phone actions)

test:
- S4 mini LTE (GT-I9195)

Commit:
code:
- separate functionality (info, phone) in layouts and toggle visibility
- added swipe to phone handler
- minor bug fixes (screen off after phone actions)

test:
- S4 mini LTE (GT-I9195)

Commit:
Merge remote-tracking branch 'origin/master'

Conflicts:
	AndroidManifest.xml
	bin/HallMonitor.apk
	res/layout/activity_default.xml
	res/values/strings.xml
	src/org/durka/hallmonitor/DefaultActivity.java
	src/org/durka/hallmonitor/Functions.java

Commit:
code:
- merge bug fixes

 test:
 - S4 mini lte (GT-I9195)

durka commented:
Oh gosh, I didn't even realize you were working on this! I'll have to study this and see which parts to integrate with my implementation. You're right, we should maybe have a wiki page for a development roadmap.

habeIchVergessen commented:
currently i'm focused on phone functionality.
- text-to-speech (80% done)
- more useful options like dismiss call with sms, mute mic, redirect to speaker (hands-free)
- display off while talking
- maybe an slide-in options-panel

Commit:
TextToSpeech implemented. Several timing problems (for instance receiving Intent ACTION_HEADSET_PLUG) made it necessary to implement in ViewCoverService.java.
for now display_name and type are used. a connected headset is required.

TODO:
- new preferences for usability

Commit:
TextToSpeech: rudimentary  preferences

Commit:
TextToSpeech: preferences for playback done.
TODO: preferences for text format

Commit:
Merge remote-tracking branch 'remotes/upstream/flashlight2'

Conflicts:
	bin/HallMonitor.apk
	res/layout/activity_default.xml
	res/values/strings.xml
	src/org/durka/hallmonitor/ConfigurationFragment.java
	src/org/durka/hallmonitor/DefaultActivity.java
	src/org/durka/hallmonitor/Functions.java

Commit:
prepare merge durka:master

Commit:
Merge branch 'master' of https://github.com/durka/HallMonitor

Conflicts:
	bin/HallMonitor.apk
	src/org/durka/hallmonitor/ConfigurationFragment.java
	src/org/durka/hallmonitor/DefaultActivity.java

Commit:
more improvements while answering calls
- update display if a call has been accepted with headset keys
- save phoneWidget states in intent and restore onResume

default widget
- show Time and Date
- notifications centered in gridView

Commit:
more improvements while answering calls
- phone widget takes over while ringing or talking when cover has been closed

Commit:
added flight mode notification

habeIchVergessen commented:
i guess that [framework branch](https://github.com/durka/HallMonitor/issues/28) contains more precise touch event handling. includes all slide stuff.
